src/main/java/my/test/demo/table/TableSortAndSearchDemo.java

```diff
@@ -0,0 +1,109 @@
+package my.test.demo.table;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.Select;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+import java.text.NumberFormat;
+import java.text.ParseException;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+
+public class TableSortAndSearchDemo {
+
+    private static final String TABLE_COLUMN_XPATH_TEMPLATE = "//table[@id='example']//tr[%s]/td[%s]";
```

Comment:
I want to share with you the following information:
findElement method can be executed on WebElement instance. In this case new element will be found inside current. F.e. (pseudo-code)
```
WebElement headerBlock = driver.findElement(By.xpath("//div[@class='header']"));
WebElement logoutLink = headerBlock.findElement(By.xpath(".//a")); //do not forget about dot for xpath in these cases
```
I don't expect that you will re-write current code, but please learn this info and use in your work